routes/cube/index.js

```diff
@@ -1418,9 +1418,14 @@ router.post('/edit/:id', ensureAuth, async (req, res) => {
   try {
     let cube = await Cube.findOne(buildIdQuery(req.params.id));
 
+    let viewQuery = 'table';
+    if (req.query.view) {
+      viewQuery = req.query.view;
+    }
+
     if (!req.user._id.equals(cube.owner)) {
       req.flash('danger', 'Only cube owner may edit.');
-      return res.redirect(`/cube/list/${encodeURIComponent(req.params.id)}`);
+      return res.redirect(`/cube/list/${encodeURIComponent(req.params.id)}?view=${viewQuery}`);
```

Comment:
Just for sake of correctness, these should probably be encoded.
```suggestion
      return res.redirect(`/cube/list/${encodeURIComponent(req.params.id)}?view=${encodeURIComponent(viewQuery)}`);
```